### ISSUE TITLE
Deactivated email address can be primary

### ISSUE THREAD

**Issue by sjjhsjjh** [ietf-tools/datatracker#7128]
### Describe the issue

Hi, I added a new email address to my IETF Datatracker account, and deactivated my old email address. Later I received a number of email messages to my old email address.
I opened the account info page for my user and noticed that no primary address was selected. (Sorry, I didn't take a screen capture at that point.) In other words, none of the radio buttons was filled in the Primay column was filled. I selected an address
![image](https://github.com/ietf-tools/datatracker/assets/25548581/4fe7bbf1-b457-4c94-b92a-51a933719bcb)
I guess what happened is that the system had retained the old email address as primary even though is was deactivated, although that wasn't reflected in the web user interface.
In summary, I think there are two problems.

- If a user has deactivated all but one of their email addresses then their primary address should default to the active one.
- The primary address, whether determined automatically or manually, should always be indicated in the web user interface.

Cheers, Jim

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by jennifer-richards** [comment]
I poked at this a little bit - deactivating the primary email and applying the change gives all indications that it succeeded, but does not actually deactivate that email address. Similarly, marking an address previously deactivated succeeds and reactivates that email address. That would all be reasonable except that the user is not given any notice about how their ambiguous request was resolved.

Automatically activating addresses or guessing at which should be primary is likely to cause user surprises - imo the Datatracker should give an error that guides the user to choosing an active address as primary. In addition to back-end enforcement, the UI should
1. Disable the "primary" toggle for inactive addresses (whether this is already applied or not)
2. Lock the checkbox for "primary" address in the "active" state until another address is chosen

I think this will prevent surprises.

**Comment by rjsparks** [comment]
Also - just to check - the messages received later were _from the datatracker_ and not from, e.g., a mailing list, correct? We do not yet have integration with the mailing list software to cause it to pay attention to the datatracker settings (though that is planned for the future).

**Comment by sjjhsjjh** [comment]
Hi, The messages were from the ietf-secretariat-reply@ietf.org address.

**Comment by sjjhsjjh** [comment]
Hi, I got some more messages to my deactivated email address, which also isn't my primary email address. I don't have too many subscriptions so I'll set up a new account and subscribe from there instead.

**Comment by rjsparks** [comment]
Send a message to support@ietf.org asking the secretariat to change the list subscriptions with your old address to the address you want.
